src/WorldStorage/NBTChunkSerializer.cpp

```diff
@@ -887,6 +888,82 @@ void cNBTChunkSerializer::AddItemFrameEntity(cItemFrame * a_ItemFrame)
 
 
 
+void cNBTChunkSerializer::AddArmorStandEntity(cArmorStand * a_ArmorStand)
+{
+	m_Writer.BeginCompound("");
+		AddBasicEntity(a_ArmorStand, "ArmorStand");
+		m_Writer.AddByte("Small", a_ArmorStand->IsSmall() ? 1 : 0);
+		m_Writer.AddByte("Gravity", a_ArmorStand->HasGravity() ? 1 : 0);
+		m_Writer.AddByte("Marker", a_ArmorStand->IsMarker() ? 1 : 0);
+		m_Writer.AddByte("Arms", a_ArmorStand->HasArms() ? 1 : 0);
+		m_Writer.AddByte("Plate", a_ArmorStand->HasBasePlate() ? 1 : 0);
+		m_Writer.AddByte("Invisible", a_ArmorStand->IsInvisible() ? 1 : 0);
+		m_Writer.AddString("CustomName", a_ArmorStand->GetCustomName());
+		m_Writer.AddByte("CustomNameVisible", static_cast<Byte>(a_ArmorStand->IsCustomNameAlwaysVisible()));
+		m_Writer.BeginList("HeadRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetHeadRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetHeadRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetHeadRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("BodyRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetBodyRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetBodyRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetBodyRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("LeftArmRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftArmRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftArmRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftArmRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("RightArmRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightArmRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightArmRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightArmRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("LeftLegRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftLegRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftLegRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetLeftLegRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("RightLegRotation", TAG_Double);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightLegRotation().x);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightLegRotation().y);
+			m_Writer.AddDouble("", a_ArmorStand->GetRightLegRotation().z);
+		m_Writer.EndList();
+		m_Writer.BeginList("Items", TAG_Compound);
```

Comment:
The wiki says this is split into "HandItems" and "ArmorItems"
https://minecraft.gamepedia.com/Armor_Stand#Data_values

Reply:
I will take care of it

Reply:
Should be solved, I then updated a part of the code to make it vanilla compliant